modules/rostests/apitests/shell32/ShellExecCmdLine.cpp

```diff
@@ -0,0 +1,409 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Test for ShellExecCmdLine
+ * PROGRAMMERS:     Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+#include "shelltest.h"
+#include <shlwapi.h>
+
+#define NDEBUG
+#include <debug.h>
+#include <stdio.h>
+
+//#define ShellExecCmdLine ShellExecCmdLine
+
+#ifndef SECL_NO_UI
+    #define SECL_NO_UI          0x2
+    #define SECL_LOG_USAGE      0x8
+    #define SECL_USE_IDLIST     0x10
+    #define SECL_RUNAS          0x40
+#endif
+
+#ifdef ShellExecCmdLine
+    static LPCWSTR Get1stArgAndParams(LPCWSTR psz, LPWSTR pszArg0, INT cchArg0)
+    {
+        LPCWSTR pch;
+        INT ich = 0;
+        if (*psz == L'"')
+        {
+            // 1st argument is quoted. the string in quotes is quoted 1st argument.
+            // [pch] --> [pszArg0+ich]
+            for (pch = psz + 1; *pch && ich + 1 < cchArg0; ++ich, ++pch)
+            {
+                if (*pch == L'"' && pch[1] == L'"')
+                {
+                    // doubled double quotations found!
+                    pszArg0[ich] = L'"';
+                }
+                else if (*pch == L'"')
+                {
+                    // single double quotation found!
+                    ++pch;
+                    break;
+                }
+                else
+                {
+                    // otherwise
+                    pszArg0[ich] = *pch;
+                }
+            }
+        }
+        else
+        {
+            // 1st argument is unquoted. non-space sequence is 1st argument.
+            // [pch] --> [pszArg0+ich]
+            for (pch = psz; *pch && !iswspace(*pch) && ich + 1 < cchArg0; ++ich, ++pch)
+            {
+                pszArg0[ich] = *pch;
+            }
+        }
+        pszArg0[ich] = 0;
+        // skip space
+        while (iswspace(*pch))
+            ++pch;
+
+        return pch;
+    }
+
+    HRESULT WINAPI ShellExecCmdLine(
```

Comment:
Please update the implementation of this function with the one you have made in shell32 (I can see some differences).

Reply:
OK. Tested on Win10 successfully.